### PR TITLE
feat(HMS-2440): private IPv4 address for GCP

### PR DESCRIPTION
**Pull request by ezr-ondrej**

### CODE REVIEW

File: scripts/rest_examples/reservation-create-gcp.http [hunk 1]
```diff
@@ -10,7 +10,6 @@ X-Rh-Identity: {{identity}}
   "image_id": "https://www.googleapis.com/compute/v1/projects/rhel-cloud/global/images/rhel-9-v20230411",
   "amount": 1,
   "machine_type": "e2-micro",
-  "launch_template_id": "{{launch_template_id}}",
```
Comment (lzap) [inline scripts/rest_examples/reservation-create-gcp.http:13]: Intentional? Just checking, might be useful.

File: internal/clients/http/gcp/gcp_client.go [hunk 2]
```diff
@@ -316,6 +316,7 @@ func (c *gcpClient) GetInstanceDescriptionByID(ctx context.Context, id, zone str
 	instanceId := strconv.FormatUint(instance.GetId(), 10)
 	instanceDesc := clients.InstanceDescription{ID: instanceId}
 	for _, n := range instance.NetworkInterfaces {
+		instanceDesc.PrivateIPv4 = *n.NetworkIP
```
Comment (lzap) [inline internal/clients/http/gcp/gcp_client.go:319]: Similarly, I would perhaps check for nil unless there is a documentation remark somewhere that this is never unallocated pointer.

**Review by lzap — APPROVED** [review]
Other than that, LGTM.

**Comment by lzap** [comment]
Can you add example similarly as I did in AWS please?

File: scripts/rest_examples/reservation-create-gcp.http (hunk 1 above)
Comment (ezr-ondrej) [reply scripts/rest_examples/reservation-create-gcp.http:13]: Just removed for testing, commited by mistake, thanks for catching that! :orange_heart:

File: internal/clients/http/gcp/gcp_client.go [hunk 3]
```diff
@@ -316,6 +316,9 @@ func (c *gcpClient) GetInstanceDescriptionByID(ctx context.Context, id, zone str
 	instanceId := strconv.FormatUint(instance.GetId(), 10)
 	instanceDesc := clients.InstanceDescription{ID: instanceId}
 	for _, n := range instance.NetworkInterfaces {
+		if n.NetworkIP != nil {
```
Comment (avitova) [inline internal/clients/http/gcp/gcp_client.go:319]: I suggest to even check for n != nil, but I do not know what documentation says.
```suggestion
		if n != nil && n.NetworkIP != nil {
```

Comment (lzap) [reply internal/clients/http/gcp/gcp_client.go:319]: It is rare to put `nil` into slice/map I would not worry too much about thi.

We have a `ptr.FromOrEmpty` generic helper, can you use that instead?

Comment (ezr-ondrej) [reply internal/clients/http/gcp/gcp_client.go]: I've used the generic helper, but did not fix the range, I guess it'd be very weird to have nil in an array.

**Review by lzap — APPROVED** [review]
I agree, @avitova?

File: internal/clients/http/gcp/gcp_client.go (hunk 3 above)
Comment (avitova) [reply internal/clients/http/gcp/gcp_client.go]: Yeah, this was just a suggestion, we do not have to check necessarily:)

**Review by avitova — APPROVED** [review]
LGTM;)

**Pull request merged**